api/filter-alerts.go

```diff
@@ -0,0 +1,228 @@
+		humiographql.FilterAlert `graphql:"updateFilterAlert(input: $input)"`
+	}
+
+	actionsIdsOrNames := make([]graphql.String, len(newFilterAlert.Actions))
+	for i, actionIdOrName := range newFilterAlert.Actions {
+		actionsIdsOrNames[i] = graphql.String(actionIdOrName)
+	}
+
+	labels := make([]graphql.String, len(newFilterAlert.Labels))
+	for i, label := range newFilterAlert.Labels {
+		labels[i] = graphql.String(label)
+	}
+
+	updateAlert := humiographql.UpdateFilterAlert{
+		ViewName:           humiographql.RepoOrViewName(viewName),
+		ID:                 graphql.String(newFilterAlert.ID),
+		Name:               graphql.String(newFilterAlert.Name),
+		Description:        graphql.String(newFilterAlert.Description),
+		QueryString:        graphql.String(newFilterAlert.QueryString),
+		ActionIdsOrNames:   actionsIdsOrNames,
+		Labels:             labels,
+		Enabled:            graphql.Boolean(newFilterAlert.Enabled),
+		RunAsUserID:        graphql.String(newFilterAlert.RunAsUserID),
+		QueryOwnershipType: humiographql.QueryOwnershipType(newFilterAlert.QueryOwnershipType),
+	}
+
+	variables := map[string]interface{}{
+		"input": updateAlert,
+	}
+
+	err := fa.client.Mutate(&mutation, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	filterAlert := mapHumioGraphqlFilterAlertToFilterAlert(mutation.FilterAlert)
+
+	return &filterAlert, nil
+}
+
+func (fa *FilterAlerts) Add(viewName string, newFilterAlert *FilterAlert) (*FilterAlert, error) {
+	if newFilterAlert == nil {
+		return nil, fmt.Errorf("newFilterAlert must not be nil")
+	}
+
+	var mutation struct {
+		humiographql.FilterAlert `graphql:"createFilterAlert(input: $input)"`
+	}
+
+	actionsIdsOrNames := make([]graphql.String, len(newFilterAlert.Actions))
+	for i, actionIdOrName := range newFilterAlert.Actions {
+		actionsIdsOrNames[i] = graphql.String(actionIdOrName)
+	}
+
+	labels := make([]graphql.String, len(newFilterAlert.Labels))
+	for i, label := range newFilterAlert.Labels {
+		labels[i] = graphql.String(label)
+	}
+
+	createFilterAlert := humiographql.CreateFilterAlert{
+		ViewName:           humiographql.RepoOrViewName(viewName),
+		Name:               graphql.String(newFilterAlert.Name),
+		Description:        graphql.String(newFilterAlert.Description),
+		QueryString:        graphql.String(newFilterAlert.QueryString),
+		ActionIdsOrNames:   actionsIdsOrNames,
+		Labels:             labels,
+		Enabled:            graphql.Boolean(newFilterAlert.Enabled),
+		RunAsUserID:        graphql.String(newFilterAlert.RunAsUserID),
+		QueryOwnershipType: humiographql.QueryOwnershipType(newFilterAlert.QueryOwnershipType),
+	}
+
+	variables := map[string]interface{}{
+		"input": createFilterAlert,
+	}
+
+	err := fa.client.Mutate(&mutation, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	filterAlert := mapHumioGraphqlFilterAlertToFilterAlert(mutation.FilterAlert)
+
+	return &filterAlert, nil
+}
+
+func (fa *FilterAlerts) Get(viewName, filterAlertName string) (*FilterAlert, error) {
```

Comment:
There is a GraphQL query for retrieving a single filter alert, but it is based on ids, and since the GO API is designed around names, this is probably the best we can do.